text/0000-router-service-computed-query-params.md

```diff
@@ -49,74 +49,78 @@ Having query params accessible on the router service would allow users to implem
 
- - Adding a query param
-    ```ts
-    import Component from "@glimmer/component";
-    import { inject as service } from '@ember/service';
-    import { action } from '@ember/object';
+By default, the query params will be serialized and deserialized via the builtin URLSearchParams API, and polyfilled for IE11.
 
-    export default class Pagination extends Component {
-      @service router;
+Should someone decide to customize how serilaization and deserializion transforms the query params, that can be done directly on the router:
 
-      @action nextPage() {
-        // `this.router.queryParams` is currently an empty object
-        const nextPage = (this.router.queryParams.page || 0) + 1
-
-        this.router.transitionTo('avengers.roster', { queryParams: { page: nextPage } });
-        // after the transition, `this.router.queryParams` is `{ page: 1 }`
-      }
+```ts
+import EmberRouter from '@ember/routing/router';
+
+import config from '../config/environment';
+
+const Router = EmberRouter.extend({
+  location: config.locationType,
+  rootURL: config.rootURL,
+
+  queryParamsConfig: {
+    serialize(queryParams: object): string {
+      // serialize object for query string
+      // default to URLSearchParams, polyfilled for IE11
+    },
+    deserialize(queryString: string): object {
+      // parse to object for use in `injectedRouter.queryString`
+      // also default to URLSerachParams
     }
-    ```
+  }
+});
+```
 
- - Removing all query params
-    ```ts
-    import Component from "@glimmer/component";
-    import { inject as service } from '@ember/service';
-    import { action } from '@ember/object';
 
-    export default class Pagination extends Component {
-      @service router;
+This will address a long standing issues from as far back as 2016,
+some new functionality for serialization and deserialization could be powered by [qs](https://www.npmjs.com/package/qs) ([3.4kb (gzip+min)](https://bundlephobia.com/result?p=qs@6.7.0)) or a lternatively, [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) -- this would enable the setting of arrays and objects which is not possible today.
 
-      @action reset() {
-        // `this.router.queryParams` is currently `{ page: 1 }`
 
-        this.router.transitionTo('avengers.roster', { queryParams: { /* empty */ } });
-        // after the transition, `this.router.queryParams` is `{ }`
-      }
-    }
-    ```
+### Sticky Query Params
 
- - Merging query params
+By default, transitionTo will clear the query params, unless specified inside the transiion.
 
-    We have previously been to the route `avengers.roster` with query params `{ page: 1 }`
-    ```ts
-    import Component from "@glimmer/component";
-    import { inject as service } from '@ember/service';
-    import { action } from '@ember/object';
+If query params are defined ahead of time as sticky, they will persist in the URL between sub routes.
```

Comment:
My interpretation of this is that sticky query params in this RFC remaning when moving between sibling routes. But query params as they work today is that they are cached and 'restored' when leaving and then re-entering the route (but not visible in between). 

If the above is correct, it seems like you are giving the word 'sticky' another meaning than it [currently has](https://guides.emberjs.com/v2.15.0/routing/query-params/#toc_sticky-query-param-values). Perhaps another word could be used, to avoid confusion?

Reply:
I'd like to argue that the current usage in the guides is incorrect, and it should be 'persistent'.

But idk how hard of a shift that'd be.

Reply:
That could very well be the case, but anyway I'd clarify that sticky and sticky means different things in this RFC and the current version of ember. To avoid confusion.

Reply:
I agree, this should be addressed. The usage here in this RFC is quite different from what we have used the term "sticky" to mean historically.